documentation/concepts.md

```diff
@@ -0,0 +1,116 @@
+|  (zone)  |                 |  (zone)  |
+|          |  ------------>  |          |
++----------+     policyB     +----------+
+```
+
+In addition to logically separating the network, zones may also apply
+firewall rules to traffic destined to the host running firewalld.
+
+Policies may apply firewall rules to a variety of traffic, e.g.
+forwarded traffic. This is useful for network level filtering.
+
+## Zone Principles
+
+Firewalld follows some strict principles in regards to zones.
+
+1. traffic ingresses one and only one zone
+2. traffic egresses one and only one zone
+3. a zone defines a level of trust
+4. intra-zone (with-in the same zone) is implicitly allowed
+5. inter-zone (zone to zone) must be explicit, denied by default
+
+Principle 4 and 5 are actually a consequence of Principle 3.
+
+Principle 4 is configurable via CLI option `--remove-forward`.
+
+## End Station Firewall
+
+When firewall rules are attached to zones, they filter traffic destined
+to the host running firewalld. This is useful for end-stations that
+provide services, e.g. a web server.
+
+This example allows `https` for clients in the internal zone.
+
+```
+# firewall-cmd --zone internal --add-service https
+```
+
+For simple end station use cases, zones provide sufficient filtering
+functionality. For more complex use cases, policies provide a generic
+abstraction that applies to all traffic patterns.
+
+## Network Firewall
+
+Policies apply firewall rules in a stateful, unidirectional manner. This
+means you only need to consider one direction of the traffic. The return
+traffic is implicitly allowed.
+
+Policies are associated with an ingress zone and an egress zone. The
+ingress zone is where the traffic originated (received). The egress zone
+is where the traffic leaves (sent).
+
+In this example, `myPolicy` will filter forwarded traffic originating
+from the `internal` ingress zone and destined for the `external` egress
+zone.
+
+```
++----------+                 +----------+
+|          |                 |          |
+| internal |     myPolicy    | external |
+|  (zone)  |  ------------>  |  (zone)  |
+|          |                 |          |
++----------+                 +----------+
+```
+
+This is a common scenario for a home network. To allow the LAN
+(internal) access the internet you can allow all traffic by setting the
+default target of the policy using `--set-target`.
+
+```
+# firewall-cmd --policy myPolicy --set-target ACCEPT
+```
+
+Alternatively, a strict network may allow only a subset of traffic. In
+this example, only `http` and `https` are allowed; other traffic will be
+rejected.
+
+```
+# firewall-cmd --policy myPolicy --add-service http
+# firewall-cmd --policy myPolicy --add-service https
+```
+
+## Runtime and Permanent Configuration
+
+Firewalld maintains separate runtime and permanent configurations. This
+allows runtime only changes. Manipulating the permanent configuration
+requires using the `--permanent` CLI option.
```

Comment:
Enhance this by: "Runtime configuration does not persist during program restarts (or machine reboots). At startup it is populated from persistent configuration." (In better English probably. ;)

Reply:
Reworded:

```
Firewalld maintains separate runtime and permanent configurations. This
allows runtime only changes. The runtime configuration does not persist
after firewalld reload or restart. At startup it is populated from the
permanent configuration. Manipulating the permanent configuration
requires using the `--permanent` option.
```